libs/sdk-ui-dashboard/src/model/commandHandlers/widgets/exportInsightWidgetHandler.ts

```diff
@@ -0,0 +1,96 @@
+// (C) 2021 GoodData Corporation
+import { SagaIterator } from "redux-saga";
+import { call, put, select } from "redux-saga/effects";
+import { serializeObjRef } from "@gooddata/sdk-model";
+import { IExecutionResult } from "@gooddata/sdk-backend-spi";
+import invariant from "ts-invariant";
+
+import { ExportInsightWidget } from "../../commands";
+import { invalidArgumentsProvided } from "../../events/general";
+import {
+    DashboardInsightWidgetExportResolved,
+    insightWidgetExportRequested,
+    insightWidgetExportResolved,
+} from "../../events/insight";
+import {
+    selectWidgetExecutionByWidgetRef,
+    selectWidgetIsExportableToCsvByWidgetRef,
+    selectWidgetIsExportableToXlsxByWidgetRef,
+    selectWidgetIsReadyForExportByWidgetRef,
+} from "../../state/widgetExecutions/widgetExecutionsSelectors";
+import { DashboardContext } from "../../types/commonTypes";
+import { createExportFunction, IExtendedExportConfig } from "@gooddata/sdk-ui";
+import { PromiseFnReturnType } from "../../types/sagas";
+
+async function performExport(
+    executionResult: IExecutionResult,
+    config: IExtendedExportConfig,
+): Promise<string> {
+    const exporter = createExportFunction(executionResult);
+    const result = await exporter(config);
+    return result.uri;
+}
+
+function* validateSettingsAndPermissions(
+    ctx: DashboardContext,
+    cmd: ExportInsightWidget,
+): SagaIterator<void> {
+    const { config, ref } = cmd.payload;
+
+    let canExport = false;
+    if (config.format === "csv") {
+        canExport = yield select(selectWidgetIsExportableToCsvByWidgetRef(ref));
+    }
+
+    if (config.format === "xlsx") {
+        canExport = yield select(selectWidgetIsExportableToXlsxByWidgetRef(ref));
```

Comment:
This is weird  that we have permission together with result. Could we have two selector for it ?